### PR TITLE
Feature/test data integration

### PR DESCRIPTION
**Pull request by tanyushanel**
CPT2108062023-1145x824
Done only data integration without any handling

### CODE REVIEW

File: src/app/models/writing-test/writing-test.component.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-import { Route } from '../../../constants/route-constant';
+import { TopicModule } from '../../interfaces/essay-speaking';
 
 @Component({
   selector: 'app-writing-test',
   templateUrl: './writing-test.component.html',
   styleUrls: ['./writing-test.component.scss'],
 })
 export class WritingTestComponent implements OnInit {
+  @Input() essay!: TopicModule | null | undefined;
```
Comment (RomPavel) [inline src/app/models/writing-test/writing-test.component.ts:11]: lets use ethier null or undefined

File: src/app/models/user-profile/user-profile.component.ts [hunk 2]
```diff
@@ -17,6 +17,8 @@ export class UserProfileComponent implements OnInit {
 
   levels = [...Object.values(Level)];
 
+  isClicked = false;
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:20]: lets name isStarted

File: src/app/models/speaking-test/speaking-test.component.ts [hunk 3]
```diff
@@ -12,6 +14,8 @@ declare let MediaRecorder: any;
   styleUrls: ['./speaking-test.component.scss'],
 })
 export class SpeakingTestComponent implements OnInit, OnDestroy {
+  @Input() speaking!: TopicModule | null | undefined;
```
Comment (RomPavel) [inline src/app/models/speaking-test/speaking-test.component.ts:17]: lets use either null or undefined

File: src/app/models/listening-test/listening-test.component.ts [hunk 4]
```diff
@@ -50,9 +47,11 @@ export class ListeningTestComponent implements OnInit, AfterViewInit, AfterViewC
...
   ];
 
-  ngOnInit(): void {
-    this.questions = [...MOCK_QUESTION_LIST_AUDITION];
-  }
+  constructor(private elementRef: ElementRef) {}
+
+  ngOnChanges(): void {}
+
+  ngOnInit(): void {}
```
Comment (RomPavel) [inline src/app/models/listening-test/listening-test.component.ts:54]: why you need ngOnInit ngOnChanges

File: src/app/models/listening-test/listening-test.component.ts [hunk 5]
```diff
@@ -1,25 +1,22 @@
...
   selector: 'app-listening',
   templateUrl: './listening-test.component.html',
   styleUrls: ['./listening-test.component.scss'],
 })
-export class ListeningTestComponent implements OnInit, AfterViewInit, AfterViewChecked {
-  constructor(private elementRef: ElementRef) {}
-
-  questions: Question[] = [];
+export class ListeningTestComponent implements OnInit, AfterViewInit, AfterViewChecked, OnChanges {
+  @Input() questions!: Question[] | undefined | null;
```
Comment (RomPavel) [inline src/app/models/listening-test/listening-test.component.ts:19]: same here

File: src/app/models/grammar-test/grammar-test.component.ts [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
 })
-export class GrammarTestComponent implements OnInit {
-  questions: Question[] = [];
+export class GrammarTestComponent implements OnInit, OnChanges {
+  @Input() questions!: Question[] | undefined | null;
 
-  ngOnInit(): void {
-    this.questions = [...MOCK_QUESTION_LIST];
-  }
+  ngOnInit(): void {}
```
Comment (RomPavel) [inline src/app/models/grammar-test/grammar-test.component.ts:12]: why you need ngOnInit ngOnChanges

File: src/app/models/grammar-test/grammar-test.component.html [hunk 7]
```diff
@@ -1 +1,2 @@
+<!-- <app-stepper [questionList]="questions"></app-stepper> -->
```
Comment (RomPavel) [inline src/app/models/grammar-test/grammar-test.component.html:1]: you need this?

File: src/app/models/user-profile/user-profile.component.ts [hunk 8]
```diff
@@ -25,11 +27,13 @@ export class UserProfileComponent implements OnInit {
     this.testStoreService.getTestResults();
   }
 
-  onStartButtonClick(): void {
+  onStartButtonClick(level: Level): void {
+    this.isClicked = true;
+
+    this.testStoreService.selectedLevel = level;
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:33]: lets add function like storeLevel(selectedLevel)

File: src/mocks/test.json [hunk 9]
```diff
@@ -0,0 +1,258 @@
+{
```
Comment (RomPavel) [inline src/mocks/test.json:1]: do we need this mock?

Comment (tanyushanel) [reply src/mocks/test.json:1]: just for seek a response. I had lack of attempts of receiving data.

File: src/app/models/writing-test/writing-test.component.ts [hunk 10]
```diff
@@ -1,19 +1,24 @@
...
-import { Route } from '../../../constants/route-constant';
+import { TopicModule } from '../../interfaces/essay-speaking';
 
 @Component({
   selector: 'app-writing-test',
   templateUrl: './writing-test.component.html',
   styleUrls: ['./writing-test.component.scss'],
 })
 export class WritingTestComponent implements OnInit {
+  @Input() essay!: TopicModule | null;
```
Comment (RomPavel) [inline src/app/models/writing-test/writing-test.component.ts:11]: in all such places lets remove exclamation mark @Input() essay: TopicModule | null = null

File: src/app/models/user-profile/user-profile.component.ts [hunk 11]
```diff
@@ -25,11 +26,15 @@ export class UserProfileComponent implements OnInit {
     this.testStoreService.getTestResults();
   }
 
-  onStartButtonClick(): void {
+  selectLevel(selected: Level): void {
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:29]: I mean we need this function in storeService

**Review by RomPavel — APPROVED** [review]

**Pull request merged**